src/HieDb/Types.hs

```diff
@@ -31,8 +32,40 @@ import Database.SQLite.Simple.FromField
 import qualified Text.ParserCombinators.ReadP as R
 
 import HieDb.Compat
+import Control.Monad (void)
+
+data HieDb = HieDb
+  { getConn :: !Connection
+  , insertModsStatement :: !(StatementFor HieModuleRow)
+  , insertRefsStatement :: !(StatementFor RefRow)
+  , insertDeclsStatement :: !(StatementFor DeclRow)
+  , insertImportsStatement :: !(StatementFor ImportRow)
+  , insertDefsStatement :: !(StatementFor DefRow)
+  , insertExportsStatement :: !(StatementFor ExportRow)
+  , insertTyperefsStatement :: !(StatementFor TypeRef)
+  , insertTypenamesStatement :: !(StatementFor (OccName, ModuleName, Unit))
+  , queryTypenamesStatement :: !(StatementFor (OccName, ModuleName, Unit))
+  , deleteInternalTablesStatement :: !(Only FilePath -> IO ())
+  }
+
+newtype StatementFor a = StatementFor Statement
+
+data NoOutput = NoOutput
+
+instance FromRow NoOutput where
+  fromRow = pure NoOutput
+
+runStatementFor_ :: ToRow a => StatementFor a -> a -> IO ()
+{-# INLINE runStatementFor_ #-}
+runStatementFor_ (StatementFor statement) params = do
+  withBind statement params $
+    void (nextRow @NoOutput statement)
 
-newtype HieDb = HieDb { getConn :: Connection }
+runStatementFor :: (ToRow a, FromRow b) => StatementFor a -> a -> IO (Maybe b)
+{-# INLINE runStatementFor #-}
+runStatementFor (StatementFor statement) params = do
```

Comment:
This doesn't seem to be used. Perhaps we should delete it. If we are to keep it, ideally the return variable wouldn't be unconstrained and we would have something like `StatementFor a b -> a -> IO b`, but we don't seem to need that functionality right now.

Reply:
I do use it in [this line](https://github.com/wz1000/HieDb/blob/daa49c059db6f0ee2df91eb2ad8def5c2dfca88c/src/HieDb/Create.hs?plain=1#L235). 

I agree on including the output variable. In a different branch where I played around, I had that exact setup, but I omitted it for simplicity's sake. Might be good to explore that if the functions in `Query.hs` get the same benefit from this treatment.